controllers/bootstrap/init.go

```diff
@@ -953,7 +955,33 @@ func CheckClusterType(mgr manager.Manager, ns string) (bool, error) {
 	}
 }
 
-func (b *Bootstrap) DeployCertManagerCR() error {
+// 1. try to get cs-ca-certificate-secret
+// 2. try to get cs-ca-certificate
+// if we get secret but not get the cert, it is BYOC
+func (b *Bootstrap) IsBYOCert() (bool, error) {
+	klog.V(2).Info("Detect if it is BYO cert")
+	secretName := "cs-ca-ceritifcate-secret"
+	secret := &corev1.Secret{}
+	err := b.Client.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: b.CSData.ServicesNs}, secret)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return false, err
+		}
+	}
```

Comment:
What happens if error is `IsNotFound`? Secret will be empty, but there is no code to check if secret is empty or not before checking for certificate object.

Reply:
Ah I didn't notice this, thanks for pointing it out